### PR TITLE
Make Ceno-Goldilocks Importable as Crate

### PR DESCRIPTION
**Pull request by darth-cy**
Naming and grammatical changes to make the `main` branch of repo importable as a regular dependency.

### CODE REVIEW

**Comment by matthiasgoergens** [comment]
> Naming and grammatical changes to make the main branch of repo importable as a regular dependency.

Hmm, but we do already import the `main` branch as a regular dependency just fine in Ceno, don't we?

File: src/extfield.rs [hunk 1]
```diff
@@ -41,10 +41,7 @@ pub trait ExtensionField:
 
     /// Convert a field elements to a u64 vector. Do not normalize it.
     fn as_non_canonical_u64_slice(&self) -> &[u64];
-
-    /// Convert self to limbs of Goldilocks elements
-    fn to_limbs(&self) -> [Self::BaseField; Self::DEGREE];
-
+
```
Comment (matthiasgoergens) [inline src/extfield.rs:44]:
```suggestion
```

File: src/extfield/fp.rs [hunk 2]
```diff
@@ -49,3 +44,10 @@ impl ExtensionField for Goldilocks {
         *b
     }
 }
+
+impl Goldilocks {
+    /// Convert self to limbs of Goldilocks elements
+    fn to_limbs(&self) -> [<Self as ExtensionField>::BaseField; <Self as ExtensionField>::DEGREE] {
```
Comment (matthiasgoergens) [inline src/extfield/fp.rs:50]: Why do we need to add a function that nobody can use?

**Review by matthiasgoergens — APPROVED** [review]
Please see my comments for some cosmetic suggestions.

Please also make a PR to Ceno to reflect the changes here: we already use the main branch here as a dependency in Ceno.

File: src/extfield.rs [hunk 3]
```diff
@@ -41,10 +41,7 @@ pub trait ExtensionField:
 
     /// Convert a field elements to a u64 vector. Do not normalize it.
     fn as_non_canonical_u64_slice(&self) -> &[u64];
-
-    /// Convert self to limbs of Goldilocks elements
-    fn to_limbs(&self) -> [Self::BaseField; Self::DEGREE];
-
```
Comment (kunxian-xia) [inline src/extfield.rs:47]: why do you want to remove this function out of trait `ExtensionField`?

Comment (matthiasgoergens) [reply src/extfield.rs:47]: I'm not even sure it matters, because the function was not implemented before nor after this PR?

File: src/lib.rs [hunk 4]
```diff
@@ -1,7 +1,6 @@
 //! This crate implements Goldilocks field with modulus 2^64 - 2^32 + 1
 //! Credit: the majority of the code is borrowed or inspired from Plonky2 with modifications.
 #![allow(incomplete_features)]
-#![feature(generic_const_exprs)]
```
Comment (matthiasgoergens) [inline src/lib.rs:4]: It's always nice, if we can remove features.

**Review by matthiasgoergens — CHANGES_REQUESTED** [review]
Sorry, we don't have a CI set up (yet), so we have to do this manually:

The tests pass, but we are getting new compiler warnings:
```
$ cargo check --all-targets
    Blocking waiting for file lock on build directory
    Checking ceno-goldilocks v0.1.0 (/home/matthias/scroll/prog/ceno-Goldilocks)
warning: method `to_limbs` is never used
  --> src/extfield/fp.rs:50:8
   |
48 | impl Goldilocks {
   | --------------- method in this implementation
49 |     /// Convert self to limbs of Goldilocks elements
50 |     fn to_limbs(&self) -> [<Self as ExtensionField>::BaseField; <Self as ExtensionField>::DEGREE] {
   |        ^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: method `to_limbs` is never used
   --> src/extfield/fp2.rs:178:8
    |
176 | impl GoldilocksExt2 {
    | ------------------- method in this implementation
177 |     /// Convert self to limbs of Goldilocks elements
178 |     fn to_limbs(&self) -> [<Self as ExtensionField>::BaseField; <Self as ExtensionField>::DEGREE] {
    |        ^^^^^^^^

warning: method `to_limbs` is never used
  --> src/extfield/fp3.rs:97:8
   |
95 | impl GoldilocksExt3 {
   | ------------------- method in this implementation
96 |     /// Convert self to limbs of Goldilocks elements
97 |     fn to_limbs(&self) -> [<Self as ExtensionField>::BaseField; <Self as ExtensionField>::DEGREE] {
   |        ^^^^^^^^

warning: `ceno-goldilocks` (lib) generated 3 warnings
warning: `ceno-goldilocks` (lib test) generated 3 warnings (3 duplicates)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 4.48s
```

Please fix these.  By contrast, `master` has no warnings.  Thanks!

(`cargo clippy --all-targets` also complains, but it already complains about `master`, so fixing that is an issue best handled separately.)

**Comment by darth-cy** [comment]
After merging in latest toolchain fix in `ceno-recursion`, the `master` branch is now importable as is and no longer require any modification. This PR therefore is closed.

**Pull request closed without merging**